File: LuaLib/Assembly-CSharp/TransformEx.lua
---@class TransformEx : System.Object
local m = {}

---@static
---@param root UnityEngine.Transform
---@param name string
---@return UnityEngine.Transform
function m.FindChild(root, name) end

---@static
---@param trans UnityEngine.Transform
---@param name string
---@param recursively boolean
---@return UnityEngine.Transform
function m.Find(trans, name, recursively) end

---@static
---@param transform UnityEngine.Transform
---@return UnityEngine.Transform
function m.RemoveAllChild(transform) end

---@static
---@param fromGameObject UnityEngine.GameObject
---@param withName string
---@return UnityEngine.GameObject
function m.GetChildGameObject(fromGameObject, withName) end

TransformEx = m
return m

File: Assets/_Complete-Game/Scripts/Core/TransformEx.cs
using UnityEngine;
using System.Collections;

[XLua.LuaCallCSharp]
public static class TransformEx
{
    public static Transform FindChild(Transform root, string name)
    {
        if (root == null)
        {
            return null;
        }

        Transform _temp = root.Find(name);
        if (_temp)
        {
            return _temp;
        }

        for (int index = 0; index < root.childCount; ++index)
        {
            Transform child = root.GetChild(index);
            _temp = FindChild(child, name);
            if (_temp)
            {
                return _temp;
            }
        }

        return _temp;
    }

    public static Transform Find(this Transform trans, string name, bool recursively)
    {
        if (!trans)
        {
            return null;
        }

        if (recursively)
        {
            return FindChild(trans, name);
        }
        else
        {
            return trans.Find(name);
        }
    }

    public static Transform RemoveAllChild(this Transform transform)
    {
        for (int index = 0; index < transform.childCount; ++index)
        {
            Transform child = transform.GetChild(index);
            GameObject.Destroy(child.gameObject);
        }

        return transform;
    }

    static public GameObject GetChildGameObject(GameObject fromGameObject, string withName)
    {
        Transform temp = FindChild(fromGameObject.transform, withName);
        if (temp != null)
            return temp.gameObject;
        return null;
    }
}
